networkx/drawing/layout.py

```diff
@@ -866,10 +869,12 @@ def rescale_layout(pos, scale=1):
         scaled positions. Each row is a position.
 
     """
+    import numpy as np
     # Find max length over all dimensions
     lim = 0  # max coordinate for all axes
     for i in range(pos.shape[1]):
-        pos[:, i] -= pos[:, i].mean()
+        center = (max(pos[:, i]) + min(pos[:, i])) / 2
```

Comment:
This probably should go to another PR. The original code offset all coordinates by the mean of all the coordinate, but if the distribution is not balanced, the center won't be the mean. I don't see why the mean is used here. Is there a particular reason to use the mean?

The change doesn't break any test either, should we add tests to cover the case?
(The previous version will break my test for bipartite layout but we might want tests dedicate to test the correctness for this function as well?)

Reply:
Hmmm...   Changing the definition of center (center of range vs center of mass) would change existing code. So, yes -- this should be in a separate PR. I don't think we've had this discussion before.

Using the mean as center (center of mass) makes "typical" points in the middle at the cost of creating some white space on one side. There's also a notion that one point should not affect the layout significantly.
Switching to center meaning midpoint (center of range) uses the entire range (no white space on any side) at the cost of sometimes having most points on one side.

There may be other reasons/arguments.  We should check how other layout engines handle it.  

And yes, this should be tested to at least encode what we do.

Can we keep it as the mean being the center for this PR and I'll open an issue to reconsider "center".